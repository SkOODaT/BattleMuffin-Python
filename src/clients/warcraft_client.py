import requests
from authlib.integrations.requests_client import OAuth2Session
from uplink import Consumer, Query, get, params, returns

from config.client_config import ClientConfiguration
from config.region_config import Locale, Region


class WarcraftClient(Consumer):
    def __init__(self, client_id: str, client_secret: str, region: Region = Region.us, locale: Locale = Locale.en_US):
        self.client_config = ClientConfiguration(client_id, client_secret, region, locale)

        oath_response = requests.get(f"{self.client_config.oauth_host}/oauth/.well-known/openid-configuration").json()
        token_endpoint = oath_response['token_endpoint']
        session = OAuth2Session(client_id, client_secret, scope="openid", token_endpoint=token_endpoint, grant_type="client_credentials")
        session.fetch_token()
        super().__init__(base_url=self.client_config.host, client=session)
        self._inject(Query("locale").with_value(self.client_config.locale.name))

    @returns.json
    @params({"namespace": "static-us"})
    @get("/data/wow/achievement-category/index")
    def get_achievement_category_index(self):
        """Returns an index of achievement categories."""
