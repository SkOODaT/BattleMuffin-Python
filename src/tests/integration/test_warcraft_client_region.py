import os
from clients.warcraft_client import WarcraftClient


def test_get_regions_index(snapshot):
    client = WarcraftClient(os.getenv("CLIENT_ID"), os.getenv("CLIENT_SECRET"))
    response = client.get_regions_index()
    assert response == snapshot


def test_get_region(snapshot):
    client = WarcraftClient(os.getenv("CLIENT_ID"), os.getenv("CLIENT_SECRET"))
    response = client.get_region(1)
    assert response == snapshot
