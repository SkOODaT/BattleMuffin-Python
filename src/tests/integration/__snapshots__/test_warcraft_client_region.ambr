# name: test_get_region
  <class 'dict'> {
    '_links': <class 'dict'> {
      'self': <class 'dict'> {
        'href': 'https://us.api.blizzard.com/data/wow/region/1?namespace=dynamic-us',
      },
    },
    'id': 1,
    'name': 'North America',
    'tag': 'US',
  }
---
# name: test_get_regions_index
  <class 'dict'> {
    '_links': <class 'dict'> {
      'self': <class 'dict'> {
        'href': 'https://us.api.blizzard.com/data/wow/region/?namespace=dynamic-us',
      },
    },
    'regions': <class 'list'> [
      <class 'dict'> {
        'href': 'https://us.api.blizzard.com/data/wow/region/1?namespace=dynamic-us',
      },
    ],
  }
---
